### DISCUSSION TITLE
Is there a way to associate an InfoMessage event with a specific SqlCommand?

### DISCUSSION THREAD

**Discussion by Metritutus** [dotnet/SqlClient#3021] [Q&A]
When receiving information from `PRINT` or informational `RAISERROR` statements, there doesn't seem to be a way to tie them back to a specific `SqlCommand`, as the `InfoMessage` event is on the `SqlConnection`, not the `SqlCommand`, and the `SqlInfoMessageEventArgs` instance doesn't appear to contain any information identifying the `SqlCommand`.

Frustratingly, it does look like the `SqlCommand` instance [is passed](https://github.com/dotnet/SqlClient/blob/7db7af14ad8b56206354ba16df8776e64f50fec3/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs#L1300) to the `FireInfoMessageEvent` method, so to my layman's eye it seems like the information isn't that far away from the creation of the `SqlInfoMessageEventArgs` instance.

Anyway, is there any way to identify which `SqlCommand` a message came from?

**Comment by paulmedynski** [comment]
This does seem like an oversight.  We will investigate further, and if appropriate, come up with an API proposal.
